app/scripts/modules/core/src/managed/ToggleManagedResourceForApplication.tsx

```diff
@@ -0,0 +1,130 @@
+import React, { memo, useState } from 'react';
+import ReactGA from 'react-ga';
+
+import { Application } from '../application';
+import { Button } from './Button';
+import {
+  Illustration,
+  IllustrationName,
+  IModalComponentProps,
+  ModalBody,
+  ModalFooter,
+  ModalHeader,
+  showModal,
+} from '../presentation';
+import { ManagedWriter } from './ManagedWriter';
+
+const logClick = (label: string, application: string) =>
+  ReactGA.event({
+    category: 'Managed Resource Environments',
+    action: `${label} clicked`,
+    label: application,
+  });
+
+export interface IToggleManagedResourceForApplicationModalProps extends IModalComponentProps {
+  application: Application;
+}
+
+interface Error {
+  data: { error: string; message: string };
+}
+
+export const MDRunningStateDescription = () => (
+  <>
+    <p>
+      <span className="bold">
+        Careful! You’re about to stop Spinnaker from managing all resources in your application.
+      </span>
+      This feature should only be used if management is not working properly and manual intervention is required.{' '}
+      <a href="https://www.spinnaker.io/guides/user/managed-delivery" target="_blank">
+        Check our documentation for more information
+      </a>
+      .
+    </p>
+    <p>
+      Do you just need to rollback?{' '}
+      <a href="https://www.spinnaker.io/guides/user/managed-delivery/pinning/" target="_blank">
+        Please use the “pin” feature instead
+      </a>
+      .
+    </p>
+  </>
+);
+
+export const MDPausedStateDescription = () => {
+  return (
+    <p>
+      You’re about to resume management of cloud resources for your application. The latest good version approved for
+      deployment will be deployed to each environment.
+    </p>
+  );
+};
+
+export const showToggleManagedResourceModal = (props: IToggleManagedResourceForApplicationModalProps) =>
+  showModal(ToggleManagedResourceForApplicationModal, props, { maxWidth: 800 });
+
+export const ToggleManagedResourceForApplicationModal = memo(
+  ({ application, closeModal, dismissModal }: IToggleManagedResourceForApplicationModalProps) => {
+    const [isSubmitting, setSubmitting] = useState<boolean>(false);
+    const [actionError, setActionError] = useState<{ error: string; message: string } | null>(null);
+    const dataSource = application.getDataSource('environments');
+    let actionToBeTaken = 'Disable';
+    let description = <MDRunningStateDescription />;
+    let icon: IllustrationName = 'disableManagement';
+
+    if (application.isManagementPaused) {
+      actionToBeTaken = 'Resume';
+      description = <MDPausedStateDescription />;
+      icon = 'runManagement';
+    }
```

Comment:
similar question as in the other component re: turning this conditional assignment into a small function.